File: code wars/6kyu/Build Tower/main.py
def tower_builder(n_floors):
    list = []
    for i in range(n_floors):
        list.append(" " * (n_floors - i - 1) + "*" * (2 * i + 1)+" " * (n_floors - i - 1))
    return list
# Could be golfed with comprehension list
def tower_builder(n):
    return [" " * (n - i - 1) + "*" * (2*i + 1) + " " * (n - i - 1) for i in range(n)]

File: code wars/6kyu/Build Tower/main.ts
export const towerBuilder = (nFloors: number): string[] => {
    const arr: Array<string> = []
    for (let i = 0; i < nFloors; i++) {
        const spaces = " ".repeat(nFloors - i - 1);
        const stars = "*".repeat(2 * i + 1);
        arr.push(spaces + stars + spaces);
    }
    return arr
}
// Could be golfed this way:
export const towerBuilderTwo = (nFloors: number): string[] => {
    return Array.from({length: nFloors}, (_, i) => `${" ".repeat(nFloors - i - 1)}${"*".repeat(2 * i + 1)}${" ".repeat(nFloors - i - 1)}`)
   }